src/commands/module.js

```diff
@@ -0,0 +1,197 @@
+import { TiError } from '../util/tierror.js';
+import { existsSync } from 'node:fs';
+import { dirname, join, parse } from 'node:path';
+import chalk from 'chalk';
+import { detect } from '../util/timodule.js';
+import { capitalize } from '../util/capitalize.js';
+
+const { bold, cyan, gray } = chalk;
+
+// if a platform is not in this map, then we just print the capitalized platform name
+const platformNames = {
+	android: 'Android',
+	commonjs: 'CommonJS',
+	iphone: 'iPhone',
+	ios: 'iOS'
+};
+
+const ModuleSubcommands = {};
+
+/**
+ * Returns the configuration for the module command.
+ * @param {Object} _logger - The logger instance
+ * @param {Object} _config - The CLI config object
+ * @param {CLI} _cli - The CLI instance
+ * @returns {Object} Module command configuration
+ */
+export function config(logger, config, cli) {
+	const subcommands = {};
+	for (const [name, subcmd] of Object.entries(ModuleSubcommands)) {
+		subcommands[name] = subcmd.conf(logger, config, cli);
+		if (subcmd.alias) {
+			subcommands[name].alias = subcmd.alias;
+		}
+	}
+	return {
+		title: 'Module',
+		defaultSubcommand: 'list',
+		skipBanner: true,
+		subcommands
+	};
+}
+
+/**
+ * Displays all installed modules.
+ * @param {Object} logger - The logger instance
+ * @param {Object} config - The CLI config object
+ * @param {CLI} cli - The CLI instance
+ * @param {Function} finished - Callback when the command finishes
+ */
+export async function run(logger, config, cli) {
+	let action = cli.command.name();
+	if (action === 'list' && cli.command.args.length) {
+		action = cli.command.args[0];
+		if (cli.argv.$_.includes('list')) {
+			throw new TiError(`Invalid argument "${action}"`, { showHelp: true });
+		}
+		cli.command = cli.command.parent;
+	}
+	for (const [name, subcommand] of Object.entries(ModuleSubcommands)) {
+		if (action === name || action === subcommand.alias) {
+			await ModuleSubcommands[name].fn(logger, config, cli);
+			return;
+		}
+	}
+	throw new TiError(`Invalid subcommand "${action}"`, { showHelp: true });
+}
+
+/**
+ * Displays a list of all installed modules.
+ * @memberof ModuleSubcommands
+ * @param {Object} _logger - The logger instance
+ * @param {Object} _config - The CLI config object
+ * @param {CLI} _cli - The CLI instance
+ * @param {Function} _finished - Callback when the command finishes
+ */
+ModuleSubcommands.list = {
+	alias: 'ls',
+	conf(_logger, _config, _cli) {
+		return {
+			desc: 'print a list of installed modules',
+			flags: {
+				json: {
+					desc: 'display installed modules as json'
+				}
+			},
+			options: {
+				output: {
+					abbr: 'o',
+					default: 'report',
+					hidden: true,
+					values: ['report', 'json']
+				},
+				'project-dir': {
+					desc: 'the directory of the project to search'
+				}
+			}
+		};
+	},
+	async fn(logger, config, cli) {
+		const isJson = cli.argv.json || cli.argv.output === 'json';
+		let projectDir;
+		let p = expand(cli.argv['project-dir'] || '.');
```

Comment:
In general: Check for nullish coalescing operators

Reply:
The `||` was intentional for those who pass in `--project-dir ""`. `||` will handle empty strings while `??` will not.